### ISSUE TITLE
In ScheduledReporter，catch Throwable will be better?

### ISSUE THREAD

**Issue by hengyunabc** [dropwizard/metrics#975]
In `ScheduledReporter`，catch `Throwable` will be better?

http://stackoverflow.com/a/24902026

``` java
    public void start(long period, TimeUnit unit) {
        executor.scheduleAtFixedRate(new Runnable() {
            @Override
            public void run() {
                try {
                    report();
                } catch (RuntimeException ex) {
                    LOG.error("RuntimeException thrown from {}#report. Exception was suppressed.", ScheduledReporter.this.getClass().getSimpleName(), ex);
                }
            }
        }, period, period, unit);
    }
```

**Comment by ryantenney** [comment]
Are you encountering an issue where the scheduled executor service stops due to uncaught exceptions?

**Comment by hengyunabc** [comment]
When I write my own reporter, I found thad I have to catch `Throwable` in `report` method, or the reporter may halt in some case. 

So I think that catch `Throwable` in `ScheduledReporter` will be better.

```java
//ScheduledReporter.java
    /**
     * Called periodically by the polling thread. Subclasses should report all the given metrics.
     *
     * @param gauges     all of the gauges in the registry
     * @param counters   all of the counters in the registry
     * @param histograms all of the histograms in the registry
     * @param meters     all of the meters in the registry
     * @param timers     all of the timers in the registry
     */
    public abstract void report(SortedMap<MetricName, Gauge> gauges,
                                SortedMap<MetricName, Counter> counters,
                                SortedMap<MetricName, Histogram> histograms,
                                SortedMap<MetricName, Meter> meters,
                                SortedMap<MetricName, Timer> timers);
```

**Comment by arteam** [comment]
Addressed by #1049
